File: 2024/Day-1/solution-1.py
import array

def populate_arrays(file_path, left_array, right_array):
    with open(file_path, 'r') as file:
        for line_number, line in enumerate(file):
            vals = line.strip().split()
            left_array.append(int(vals[0]))
            right_array.append(int(vals[1]))

def calculate_distance(left_array, right_array):
    total_distance = 0

    for i in range(len(left_array)):
        distance = left_array[i] - right_array[i]
        total_distance += abs(distance)
    
    return total_distance

## Initalize and run

# sample_file = "Inputs/sample.txt"
real_file = "Inputs/input-1.txt"

unsorted_left_array = []
unsorted_right_array = []

sorted_left_array = []
sorted_right_array = []

total_distance = 0

# initalize arrays from input file
populate_arrays(real_file, unsorted_left_array, unsorted_right_array)

# sort each array
sorted_left_array = sorted(unsorted_left_array)
sorted_right_array = sorted(unsorted_right_array)

# calculate total distance
total_distance = calculate_distance(sorted_left_array, sorted_right_array)

# print total distance
print(total_distance)

File: 2024/Day-1/solution-1.rb
# frozen_string_literal: true

require 'csv'

## Read inputs for Arrays, "Left" and "Right"
def populateArrays(input_file, left_array, right_array)
  File.foreach(input_file) do |line|
    values = line.split(/\s+/)
    left_array << values[0].to_i
    right_array << values[1].to_i
  end
end

def calculateDistance(sorted_left_array, sorted_right_array)
  total_distance = 0

  sorted_left_array.size.times do |i|
    distance = sorted_left_array[i] - sorted_right_array[i]
    total_distance += distance.abs
  end

  total_distance
end

## Initalize and run

# sample_file = "Inputs/sample.txt"
real_file = 'Inputs/input-1.txt'

unsorted_left_array = []
unsorted_right_array = []

# initalize arrays from input file
populateArrays(real_file, unsorted_left_array, unsorted_right_array)

# sort each array
sorted_left_array = unsorted_left_array.sort
sorted_right_array = unsorted_right_array.sort

# calculate total distance
total_distance = calculateDistance(sorted_left_array, sorted_right_array)

# print total distance
puts total_distance
